### PR TITLE
Reduce lost miner stake to MIN_STAKE

### PR DESCRIPTION
**Pull request by kronosapiens**
Follow-on to #397 which reduces lost stake to MIN_STAKE (instead of entire stake)

### CODE REVIEW

File: contracts/TokenLocking.sol [hunk 1]
```diff
@@ -91,14 +91,14 @@ contract TokenLocking is TokenLockingStorage, DSMath {
   // This function is only used in context of reputation mining
   // TODO: After we add formula to calculate user's loss, refactor accordingly and/or
```
[the remark this answers is not in the document]
Comment (kronosapiens) [reply contracts/TokenLocking.sol]: Not that I noticed but there I may have just been unaware.

File: contracts/TokenLocking.sol [hunk 2]
```diff
@@ -91,14 +91,14 @@ contract TokenLocking is TokenLockingStorage, DSMath {
...
-  function punishStakers(address[] _stakers, address _beneficiary) public onlyReputationMiningCycle {
+  function punishStakers(address[] _stakers, address _beneficiary, uint256 _amount) public onlyReputationMiningCycle {
     address clnyToken = IColony(IColonyNetwork(colonyNetwork).getMetaColony()).getToken();
     uint256 lostStake;
     // Passing an array so that we don't incur the EtherRouter overhead for each staker if we looped over
     // it in ReputationMiningCycle.invalidateHash;
     for (uint256 i = 0; i < _stakers.length; i++) {
-      lostStake = userLocks[clnyToken][_stakers[i]].balance;
-      userLocks[clnyToken][_stakers[i]].balance = 0;
+      lostStake = min(userLocks[clnyToken][_stakers[i]].balance, _amount);
```
[the remark this answers is not in the document]
Comment (kronosapiens) [reply contracts/TokenLocking.sol]: I thought about that as well, I don't think it's possible for the balance to be less than MIN_STAKE but I figured there was no reason not to use the more secure pattern?

File: contracts/TokenLocking.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (elenadimitrova) [reply contracts/TokenLocking.sol]: Let's remove this then. I can't justify moving any more of the logic out of `TokenLocking` and into `ColonyNetworkMining` as the note suggests. This was done back when Token locking was implemented with the idea that potentially `punishStakers` logically should live in the same contract as `rewardStakers`

File: contracts/TokenLocking.sol (hunk 2 above)
[the remark this answers is not in the document]
Comment (elenadimitrova) [reply contracts/TokenLocking.sol]: Maybe it's a scenario if the minimum stake was increased (via a contract upgrade), post someone having staked the minimum? I agree it's a good and required safety net just very difficult to test.

**Pull request merged**